tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/core/portal/BundleSupervisor.java

```diff
@@ -130,4 +150,26 @@ public boolean stdout( String out ) throws Exception
         return true;
     }
 
+    public void updateHostBundle( String fragmentHostName, BundleDTO[] existingBundles ) throws Exception
+    {
+        long fragmentHostId = -1;
+
+        for( BundleDTO bundle : existingBundles )
+        {
+            if( bundle.symbolicName.equals( fragmentHostName ) )
+            {
+                fragmentHostId = bundle.id;
+                break;
+            }
+        }
+
+        if( fragmentHostId > 0 )
+        {
+            Agent agent = getAgent();
+            agent.redirect( Agent.COMMAND_SESSION );
+            agent.stdin( "update " + fragmentHostId );
```

Comment:
Hey @AndyWu2015  why can't you use the ```agent.update(...)``` api instead of having to pass this string?